trillian/ctfe/handlers.go

```diff
@@ -770,6 +796,10 @@ func getEntries(ctx context.Context, li *logInfo, w http.ResponseWriter, r *http
 		if leaf.LeafIndex != start+int64(i) {
 			return http.StatusInternalServerError, fmt.Errorf("backend returned unexpected leaf index: rsp.Leaves[%d].LeafIndex=%d for range [%d,%d]", i, leaf.LeafIndex, start, end)
 		}
+
+		if err := li.issuanceChainService.FixLogLeaf(ctx, leaf); err != nil {
+			return http.StatusInternalServerError, fmt.Errorf("failed to fix log leaf: %v", rsp)
```

Comment:
Following up on our conversation, pick one of the the two options:
 1. check whether the feature is enabled outside of `issuanceChainService`, and then call it or not, everywhere. Right now the behavior for `addChainInternal` is different than the one for `getEntries`.
 2. always gall `issuanceChainService` and leave all the business logic in there, as Martin suggests

Reply:
The code is now refactored to always call the `issuanceChainService` to build and fix the log leaf.